packages/next-auth/src/next/middleware.ts

```diff
@@ -101,7 +101,7 @@ async function handleMiddleware(
   options: NextAuthMiddlewareOptions | undefined,
   onSuccess?: (token: JWT | null) => Promise<NextMiddlewareResult>
 ) {
-  const { pathname, search, origin } = req.nextUrl
+  const { pathname, search, origin, basePath: nextJsBasePath } = req.nextUrl
```

Comment:
I suggest we use the variable name`basePath` here. I don't want to use `nextJsBasePath` since we're in `next` specific logic already.
To avoid conflict, we could rename the existing `basePath` at line 108 to `authPath` 🤔 this value is the pathname of the `NEXTAUTH_URL`.